runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowWorkerHarnessHelper.java

```diff
@@ -95,6 +97,17 @@ public static void configureLogging(DataflowWorkerHarnessOptions pipelineOptions
     DataflowWorkerLoggingInitializer.configure(pipelineOptions);
   }
 
+  public static void runUserDefinedInitialization() {
+    ServiceLoader<DataflowWorkerInitializer> loader =
```

Comment:
Sort these so there is a stable order based upon the full qualified class name so if the user rebuilds their application they should get a stable execution order across runs.

Also use the ReflectHelper to find the class loader.

You could copy most of the code from here: https://github.com/apache/beam/blob/1d9daf1aca101fa5a194cbbba969886734e08902/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java#L1793

Reply:
Do you think it would make sense to add a helper to ReflectHelpers like `loadServicesOrdered(clazz)` for this pattern?

Reply:
Sounds worthwhile. Since it has been copied a few times now.

Reply:
Done.